sdk/cosmos/azure-cosmos/azure/cosmos/_retry_utility.py

```diff
@@ -112,6 +128,33 @@ def Execute(client, global_endpoint_manager, function, *args, **kwargs):
                 retry_policy = sessionRetry_policy
             elif exceptions._partition_range_is_gone(e):
                 retry_policy = partition_key_range_gone_retry_policy
+            elif exceptions._container_recreate_exception(e):
+                retry_policy = container_recreate_retry_policy
+                # pylint: disable=protected-access
+                # Before we retry if retry policy is container recreate, we need refresh the cache of the
+                # container properties and pass in the new RID in the headers.
+                client._refresh_container_properties_cache(retry_policy.container_link)
+                if retry_policy.check_if_rid_different(
+                        retry_policy.container_link, client._container_properties_cache, retry_policy.container_rid):
+                    retry_policy.refresh_container_properties_cache = False
```

Comment:
if we're already dealing with the container recreate exception, is there any case in which this check will happen? won't it always be different?

Reply:
There are two cases in which it will be different. The first is if there is no throughput properties in the container and it is not due to a container recreate, in which case we don't want to retry and just let it fail. The second is if it failed to properly extract the partition key from the document definition and it was not due to a container recreate, in which case we also want to let it fail without a retry.

Reply:
Worth documenting these branches from the perspective of the exception using Mermaid and attaching it to the PR description - will help while referencing this PR in the future:
1. What happens when 404 `OWNER_RESOURCE_NOT_FOUND`
2. What happens when 404 `COLLECTION_RID_MISMATCH`
3. What happens when 404 `THROUGHPUT_OFFER_NOT_FOUND`

Reply:
@jeet1995 Thank you for the suggestion, I added a Mermaid diagram.